### ISSUE TITLE
Scan and remat the baseline axis in RiemannVis to bound the reference kernel's peak memory

### ISSUE THREAD

**Issue by chrisfinlay** [epfl-radio-astro/tabascal#114]
`RiemannVis` (the non-FFI RFI-visibility component) materialises a full
`(n_bl, n_rfi, n_freq_fine, n_time_fine)` intermediate before reducing it. Scanning
the baseline axis under `checkpoint` would bound that, the way #108 did for the
antenna axis in the Fourier RFI-signal component.

## Where the memory goes

`interferometry.calculate_rfi_vis_fine`:

```python
vis_rfi_fine = jnp.sum(
    rfi_A_[a1]
    * jnp.conjugate(rfi_A_[a2])
    * jnp.exp(1.0j * (rfi_phase_[a1] - rfi_phase_[a2])),
    axis=1,
)
```

`rfi_A_[a1]` gathers `(n_ant, n_rfi, ...)` up to `(n_bl, n_rfi, n_freq_fine, n_time_fine)`, and the product and `exp` each produce another array that shape. Only then does `sum(axis=1)` collapse the RFI axis, and only after that does the reshape/mean take it down to the coarse `(n_bl, n_freq, n_time)` grid. The peak is set by an intermediate that is `n_rfi * (n_int_freq * n_int_time)` times larger than the result.

That is precisely the term the FFI kernel exists to avoid — it fuses gather, multiply, sum and the fine->coarse reduction into one pass. The cost of not doing so, from the re-measured perf references:

| variant | peak, double | peak, single |
|---|---|---|
| `RiemannVis` | **16.887 GB** | **8.679 GB** |
| `RiemannVisFFI` | 1.239 GB | 0.606 GB |

~13.6x in double, ~14.3x in single.

## Why it still matters, given the FFI kernel exists

`RiemannVis` is not just a fallback — it is the **reference implementation the FFI kernel is validated against**, in value (`test_ffi_matches_reference`), forward mode (`test_ffi_jvp`) and reverse mode (`test_ffi_vjp`), all in `tests/components/test_rfi_vis.py`.

So its memory ceiling caps the sizes at which the kernel under test can be checked at all. #106 recorded exactly this: on a 6 GB GTX 1060, 4 tests OOM at the largest `test_ffi_jvp`/`test_ffi_vjp` size in fp64, where the reference intermediate alone is 945 MiB and JVP/VJP hold several live. **The reference blows up before the thing it is meant to validate does**, which is backwards.

It is also still a selectable component and one of the two variants the CSCS perf check runs.

## Proposal

Scan the baseline axis, with `checkpoint` on the body, chunked by a `batch_size` knob so the memory/launch-overhead trade is tunable rather than fixed at one baseline per step.

Two things make this a better shape than #108's antenna scan:

- **No transpose.** The output is `(n_bl, n_freq, n_time)` with the scanned axis already leading, so `lax.scan`'s axis-0 stacking matches the result layout directly. #108 needed a `swapaxes` back, which it flagged as recoverable overhead; there is nothing equivalent here.
- **The reduction is already outside.** The fine->coarse mean happens per body, so each step emits coarse-grid output and the stacked residuals stay small.

`checkpoint` is load-bearing rather than decorative, for the same reason as #108: `lax.scan` stacks the body's residuals across iterations for reverse-mode AD, so without remat the scan would fix the forward peak and leave the tape roughly where it was. That matters more here than it did there, because reverse mode is explicitly under test (`test_ffi_vjp`).

## Caveats to check rather than assume

- **XLA may already fuse this on CPU.** An earlier look at the same function suggested the CPU backend fuses the gather-multiply-sum chain such that the intermediate is never materialised, in which case a CPU benchmark would show no win and the change has to be verified on GPU. Measure there before concluding anything.
- **Sharding must keep working.** The body runs inside `psum_over_rfi(local_vis)` under `shard_map`; the scan goes inside `local_vis`, and the fine->coarse mean must stay before the cross-device sum (they commute, which is what keeps the collective coarse-grid sized).
- **This is not a speedup.** #108's equivalent cost ~5% runtime for 2.45x memory. The goal here is to make the reference implementation runnable at the sizes the FFI kernel needs validating at, not to make it fast.
- **It would move the `Riemann` perf references.** The memory numbers re-measured in #106 would need re-recording for both `Riemann` variants; the `RiemannFFI` ones are untouched.

## Related

- #108 — the same scan+remat treatment applied to the antenna axis, with measurements
- #107 — where the non-FFI path was identified as the dominant memory term before the FFI kernel removed it
- #106 — the re-measured references this would change, and the GTX 1060 OOM note
